src/test/java/taub/citi/CitiBikeServiceTest.java

```diff
@@ -0,0 +1,155 @@
+    {
+        String currStationId = "1846085734612252774";
+        Stations stations = service.getStatuses().blockingGet();
+        Station currStation = stations.data.stations.get(0);
+        for (Station station : stations.data.stations)
+        {
+            if (station.station_id.equals(currStationId))
+            {
+                currStation = station;
+                break;
+            }
+        }
+        assertEquals(currStation.station_id, currStationId);
+    }
+
+    @Test
+    void getClosestAvailableBikeLocation()
+    {
+        double currLat = 40.5;
+        double currLon = -73.1;
+        Stations stationInfo = service.getStations().blockingGet();
+        Stations stationStatus = service.getStatuses().blockingGet();
+        int retval = 1; // finding the closest bike
+        List<Station> combinedStations =
+                combineStationInfo(stationInfo.data.stations, stationStatus.data.stations, retval);
+        Station station = findClosestStation(currLat, currLon, combinedStations);
+        assertNotEquals(0, station.num_bikes_available);
+    }
+
+    @Test
+    void getClosestAvailableSlotLocation()
+    {
+        double currLat = 40.5;
+        double currLon = -73.1;
+        Stations stationInfo = service.getStations().blockingGet();
+        Stations stationStatus = service.getStatuses().blockingGet();
+        int retval = 2; // finding the closest slot
+        List<Station> combinedStations =
+                combineStationInfo(stationInfo.data.stations, stationStatus.data.stations, retval);
+        Station station = findClosestStation(currLat, currLon, combinedStations);
+        assertNotEquals(0, station.num_docks_available);
+    }
+
+    private List<Station> combineStationInfo(List<Station> stationInfo, List<Station> stationStatus, int retVal)
+    {
+        stationInfo.sort(Comparator.comparing(Station::getStationId));
+        stationStatus.sort(Comparator.comparing(Station::getStationId));
+        List<Station> combinedStations = new ArrayList<>();
+        for (int ix = 0; ix < stationStatus.size(); ix++)
+        {
+            Station station = stationStatus.get(ix);
+            station.lat = stationInfo.get(ix).lat;
+            station.lon = stationInfo.get(ix).lon;
+
+            // if finding the closest bike, omit stations with no bikes available
+            if (retVal == 1)
+            {
+                if (station.num_bikes_available != 0)
+                {
+                    combinedStations.add(station);
+                }
+            } else if (retVal == 2) // if finding the closest dock, omit stations with no docks available
+            {
+                if (station.num_docks_available != 0)
+                {
+                    combinedStations.add(station);
+                }
+            }
+        }
+        return combinedStations;
+    }
+
+    public Station findClosestStation(double lat, double lon, List<Station> stations)
```

Comment:
This code should not be in your test. You will need to use this later in your lambda.